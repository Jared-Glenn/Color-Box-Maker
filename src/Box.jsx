import React from "react";

function Box({ color, width, height, deleteBox }) {
    return (
        <>
            <div style={{backgroundColor: color, width: `${width}px`, height: `${height}px`}} data-testid="box"></div>
            <button onClick={ deleteBox }>X</button>
        </>
    )
};

export default Box;