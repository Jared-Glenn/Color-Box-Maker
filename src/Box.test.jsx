import { render } from "@testing-library/react";
import Box from "./Box";

// Smoke Test
it('renders without crashing', () => {
    render(<Box
            key={0}
            color={"pink"}
            width={"100"}
            height={"100"}
            />);
});

// Snapshot Test
it("matches the snapshot", () => {
    const {asFragment} = render(<Box
                                    key={0}
                                    color={"pink"}
                                    width={"100"}
                                    height={"100"}
                                    />);
                                    expect(asFragment()).toMatchSnapshot();
});
