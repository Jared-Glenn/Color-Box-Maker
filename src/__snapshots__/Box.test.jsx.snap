// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="box"
    style="background-color: pink; width: 100px; height: 100px;"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
